src/utilities/lexicographicSortSchema.js

```diff
@@ -74,6 +75,20 @@ export function lexicographicSortSchema(schema: GraphQLSchema): GraphQLSchema {
     return maybeType && replaceNamedType(maybeType);
   }
 
+  function replaceResolvedType(
+    maybeTypeResolver: ?GraphQLTypeResolver<mixed, mixed>,
+  ) {
+    return (
+      maybeTypeResolver &&
+      ((...args) =>
+        Promise.resolve(maybeTypeResolver(...args)).then((resolvedType) =>
+          isObjectType(resolvedType)
```

Comment:
Just return `resolvedType.name` is should be simpler and faster.
Also by using `replaceNamedType` you making reference to the previous schema so it can't be garbage collected.